File: backend/services/ai/src/main/java/app/mnema/ai/client/media/MediaResolved.java
package app.mnema.ai.client.media;

import java.time.Instant;
import java.util.UUID;

public record MediaResolved(
        UUID mediaId,
        String kind,
        String url,
        String mimeType,
        Long sizeBytes,
        Integer durationSeconds,
        Integer width,
        Integer height,
        Instant expiresAt
) {
}

File: backend/services/user/src/main/kotlin/app/mnema/user/media/client/MediaResolved.kt
package app.mnema.user.media.client

import java.time.Instant
import java.util.UUID

data class MediaResolved(
    val mediaId: UUID,
    val kind: String? = null,
    val url: String? = null,
    val mimeType: String? = null,
    val sizeBytes: Long? = null,
    val durationSeconds: Int? = null,
    val width: Int? = null,
    val height: Int? = null,
    val expiresAt: Instant? = null
)
